qiling/os/utils.py

```diff
@@ -78,19 +78,53 @@ def convert_posix_to_win32(rootfs, cwd, path):
         cwd = PurePosixPath(cwd[1:])
         path = PurePosixPath(path)
         if path.is_absolute():
-            return rootfs / path.relative_to(path.anchor)
+            return rootfs / PathUtils.normalize(path)
         else:
-            return rootfs / cwd / path
+            return rootfs / PathUtils.normalize(cwd / path)
     
     @staticmethod
     def convert_for_native_os(rootfs, cwd, path):
         rootfs = Path(rootfs)
         cwd = PurePosixPath(cwd[1:])
         path = Path(path)
         if path.is_absolute():
-            return rootfs / path.relative_to(path.anchor)
+            return rootfs / PathUtils.normalize(path)
         else:
-            return rootfs / cwd / path
+            return rootfs / PathUtils.normalize(cwd / path.as_posix())
+
+    @staticmethod
+    def normalize(path):
+        sep = path._flavour.sep
```

Comment:
Quite tricky here. But it seems that we can't find a better approach. I suggest adding a dict like:

```python
seps = {
PureWindowsPath : "\\"
...
}
```